base/compiler/tfuncs.jl

```diff
@@ -287,6 +287,11 @@ function isdefined_tfunc(@nospecialize(arg1), @nospecialize(sym))
             elseif a1.name === _NAMEDTUPLE_NAME
                 if isconcretetype(a1)
                     return Const(false)
+                else
+                    ns = a1.parameters[1]
+                    if isa(ns, Tuple)
+                        return Const(val in ns)
```

Comment:
So this happens when the NamedTuple is not concrete (e.g. because the types are not known), but the field names are known? Seems reasonable to me, but please add a test that tests this codepath (i.e. fails without this commit and passes with it).

Reply:
> So this happens when the NamedTuple is not concrete (e.g. because the types are not known), but the field names are known?

yeah, exactly.

I will try to add a test later, I just submitted this without a test because I wanted to hear if this is acceptable change first.

Reply:
done

Reply:
This should use `1 <= idx <= length(ns)`.

Reply:
nice, applied in 8823625